File: lib/04_mock.rb
class Vault
  # 🙅‍♀️ DO NOT CHANGE THIS CODE - START

  def initialize(secret, authoriser)
    @secret = secret
    @authoriser = authoriser
  end

  # 🙅‍♀️ DO NOT CHANGE THIS CODE - END

  def get(username, password)
    # ✍️ WRITE YOUR CODE HERE

    # need to throw an error?
    # if condition
    #   raise "not authorized"
    # end

    @secret
  end
end

File: javascript/src/04_mock.js
class Vault {
  constructor(secret, authoriser) {
    this.secret = secret;
    this.authoriser = authoriser;
  }

  get(username, password) {
    // WRITE YOUR CODE HERE

    // need to throw an error?
    //
    // if (condition) {
    //   throw new Error("oh no");
    // }

    return this.secret;
  }
}

module.exports = Vault;
